### PR TITLE
refactor(trx_mngr): refactor processing of block trxs

### PR DESCRIPTION
**Pull request by MatusKysel**
## Purpose

While working on double verification I have noticed that processing of block trxs is kind of mess. This is just small refactor of it

### CODE REVIEW

File: src/network/taraxa_capability.cpp [hunk 1]
```diff
@@ -251,13 +251,13 @@ void TaraxaCapability::interpretCapabilityPacketImpl(NodeID const &_nodeID, unsi
         }
 
         if (network_id != conf_.network_id) {
-          LOG(log_er_) << "Incorrect network id " << network_id << ", host " << _nodeID << " will be disconnected";
+          LOG(log_wr_) << "Incorrect network id " << network_id << ", host " << _nodeID << " will be disconnected";
```
Comment (JakubFornadel) [inline src/network/taraxa_capability.cpp:254]: We could use _nodeID.abridged() in these logs

File: src/transaction_manager/transaction_queue.cpp [hunk 2]
```diff
@@ -92,43 +92,56 @@ void TransactionQueue::removeTransactionFromBuffer(trx_hash_t const &hash) {
 }
 
 void TransactionQueue::addTransactionToVerifiedQueue(trx_hash_t const &hash, std::list<Transaction>::iterator iter) {
-  uLock lock(shared_mutex_for_verified_qu_);
+  uLock verify_lock(shared_mutex_for_verified_qu_);
+  {
+    // this can happend when tx is verified in backround thread
+    // and in same time we verified block, so it was already removed
```
Comment (JakubFornadel) [inline src/transaction_manager/transaction_queue.cpp:98]: hmm I would not really know what this is about if we did not have personal convo about it...

File: src/transaction_manager/transaction_manager.cpp [hunk 3]
```diff
@@ -412,28 +350,71 @@ void TransactionManager::packTrxs(vec_trx_t &to_be_packed_trx, uint16_t max_trx_
...
+  std::set<trx_hash_t> known_trx_hashes(all_block_trx_hashes.begin(), all_block_trx_hashes.end());
+
+  auto trx_batch = db_->createWriteBatch();
   for (auto const &trx : trxs) {
-    auto valid = verifyTransaction(trx);
-    if (!valid.first) {
-      invalidTransaction = true;
-      LOG(log_er_) << "Invalid transaction " << trx.getHash().toString() << " " << valid.second;
+    known_trx_hashes.erase(trx.getHash());
+    auto status = db_->getTransactionStatus(trx.getHash());
```
Comment (JakubFornadel) [inline src/transaction_manager/transaction_manager.cpp:367]: Lets optimize this one. You already have vector of all txs hashes, there is a method to get all txs statuses from db at once and parameter is vector of hashes...

Comment (JakubFornadel) [reply src/transaction_manager/transaction_manager.cpp:367]: hmm okay you dont have the vector of hashes, but might be still worth to loop over txs first and get hashes + 1 db call instead off N db calls.... ???

**Review by JakubFornadel — APPROVED** [review]

File: src/transaction_manager/transaction_manager.cpp (hunk 3 above)
Comment (MatusKysel) [reply src/transaction_manager/transaction_manager.cpp:367]: but status can be outdated as verification of all unverified trxs is done in background by other threads, so this way as it is you will get as much up to date status as possible. And one more thing reading from DB is cheap, re-validating is expensive

File: src/network/taraxa_capability.cpp [hunk 4]
```diff
@@ -251,13 +251,13 @@ void TaraxaCapability::interpretCapabilityPacketImpl(NodeID const &_nodeID, unsi
...
           host->disconnect(_nodeID, p2p::UserReason);
           break;
         }
 
         if (genesis_hash != dag_mgr_->get_genesis()) {
-          LOG(log_er_) << "Incorrect genesis hash " << genesis_hash << ", host " << _nodeID << " will be disconnected";
+          LOG(log_wr_) << "Incorrect genesis hash " << genesis_hash << ", host " << _nodeID << " will be disconnected";
```
Comment (gaoqi8649) [inline src/network/taraxa_capability.cpp:260]: Same reason as above

File: src/network/taraxa_capability.cpp [hunk 5]
```diff
@@ -242,7 +242,7 @@ void TaraxaCapability::interpretCapabilityPacketImpl(NodeID const &_nodeID, unsi
         // We need logic when some different node versions might still be compatible
         if (node_major_version != FullNode::c_node_major_version ||
             node_minor_version != FullNode::c_node_minor_version) {
-          LOG(log_er_) << "Incorrect node version: " << getFormattedVersion(node_major_version, node_minor_version)
+          LOG(log_wr_) << "Incorrect node version: " << getFormattedVersion(node_major_version, node_minor_version)
```
[the remark this answers is not in the document]
Comment (MatusKysel) [reply src/network/taraxa_capability.cpp:245]: OK

**Review by mfrankovi — APPROVED** [review]

**Pull request merged**